### PR TITLE
test(e2e): Add test for updated `boundary authenticate` command

### PR DESCRIPTION
**Pull request by moduli**
Note: This was an experiment to see what was possible. The usefulness of this test is still up for debate.

In https://github.com/hashicorp/boundary/pull/2725, there is a new "short-hand" version of the `authenticate` command that allows users to more quickly log in to boundary. Now, users can just invoke `boundary authenticate`, and boundary will use the user's primary auth method to log in. This PR adds an e2e test to validate this behavior.

Originally, the test tried to use `exec.Cmd.Stdin` to provide the login credentials when `boundary authenticate` prompts for a username and password, but that didn't seem to work. @johanbrandhorst pointed me to try using the `creack/pty` package.

**Test Result**
```
❯ go test github.com/hashicorp/boundary/testing/internal/e2e/tests/static -v -run TestCliAuthenticatePrimary -count=1
=== RUN   TestCliAuthenticatePrimary
    scope.go:57: Created Org Id: o_r9agMdLs5l
--- PASS: TestCliAuthenticatePrimary (1.87s)
PASS
ok      github.com/hashicorp/boundary/testing/internal/e2e/tests/static 2.129s
```

**Test Result: Intentionally failing by providing the wrong password**
```
❯ go test github.com/hashicorp/boundary/testing/internal/e2e/tests/static -v -run TestCliAuthenticatePrimary -count=1
=== RUN   TestCliAuthenticatePrimary
    authenticate_primary_test.go:58: 
                Error Trace:    /Users/mycow/Developer/boundary/testing/internal/e2e/tests/static/authenticate_primary_test.go:58
                Error:          Not equal: 
                                expected: 200
                                actual  : 0
                Test:           TestCliAuthenticatePrimary
                Messages:       {"context":"Error from controller when performing authentication","status":401,"api_error":{"kind":"Unauthenticated","message":"Unable to authenticate."}}
--- FAIL: TestCliAuthenticatePrimary (1.32s)
FAIL
FAIL    github.com/hashicorp/boundary/testing/internal/e2e/tests/static 1.712s
FAIL
```

### CODE REVIEW

File: testing/internal/e2e/tests/static/authenticate_primary_test.go [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+	require.NoError(t, err)
+	t.Cleanup(func() {
+		err := f.Close()
+		require.NoError(t, err)
+	})
+
+	// Added to prevent inputting login credentials before the authenticate command prompts for
+	// username/password. Without this, the test would print the username and password onto the
+	// terminal as opposed to being hidden by the prompt.
+	time.Sleep(500 * time.Millisecond)
```
Comment (moduli) [inline testing/internal/e2e/tests/static/authenticate_primary_test.go:43]: Is there a better way to handle this?

Comment (johanbrandhorst) [reply testing/internal/e2e/tests/static/authenticate_primary_test.go:43]: As discussed on Slack, we should try to read from the pty.

File: testing/internal/e2e/tests/static/authenticate_primary_test.go [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+	f, err := pty.Start(cmd)
+	require.NoError(t, err)
+	t.Cleanup(func() {
+		err := f.Close()
+		require.NoError(t, err)
+	})
+
+	f.Write([]byte(bc.AdminLoginName + "\n"))
+	f.Write([]byte(bc.AdminLoginPassword + "\n"))
+	f.Write([]byte{4}) // EOT (End of Transmission - marks end of file stream)
```
Comment (johanbrandhorst) [inline testing/internal/e2e/tests/static/authenticate_primary_test.go:41]: :just_right:

**Review by johanbrandhorst — APPROVED** [review]
